WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReaders/CardReaderSettingsViewController.swift

```diff
@@ -1,9 +1,91 @@
 import UIKit
+import Combine
+
+final class CardReaderSettingsViewController: UIViewController {
+    private var subscriptions = Set<AnyCancellable>()
+    private var viewmodel = CardReaderSettingsViewModel()
+    private var subView: UIView?
+    private var alert: UIAlertController?
 
-class CardReaderSettingsViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         configureNavigation()
+
+        self.viewmodel = CardReaderSettingsViewModel()
+        viewmodel.$summaryState
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] summaryState in
+                self?.setSubView()
+            }
+            .store(in: &subscriptions)
+    }
+
+    private func setSubView() {
+        alert?.dismiss(animated: true, completion: nil)
+        subView?.removeFromSuperview()
+
+        switch self.viewmodel.summaryState {
+        case .cleanSlate:
+            addCleanSlateView()
+        case .connected:
+            addListView()
+        case .notConnected:
+            addListView()
+        }
+
+        switch self.viewmodel.interactiveState {
+        case .none:
+            noop()
+        case .searching:
+            addSearchingModal()
+        case .foundReader:
+            noop()
+        case .connecting:
+            noop()
+        case .tutorial:
+            noop()
+        case .updateAvailable:
+            noop()
+        case .updateRequired:
+            noop()
+        case .updating:
+            noop()
+        }
+    }
+
+    private func addCleanSlateView() {
+        let connectView = CardReaderSettingsConnectView(frame: view.frame)
+        connectView.onPressedConnect = {
+            self.viewmodel.startSearch()
+        }
+        subView = connectView
+        if subView != nil {
+            view.addSubview(subView!)
+        }
+    }
+
+    private func addListView() {
+        // TODO Implement
+    }
+
+    private func noop() {
+        // TODO cleanup
+    }
+
+    private func addSearchingModal() {
+        // TODO Use FancyAlert instead
```

Comment:
Oh, ok. I should have reached this point before commenting about the FancyAlertViewController 😊